libcoap/src/context.rs

```diff
@@ -406,6 +410,32 @@ impl CoapContext<'_> {
         self.add_endpoint(addr, coap_proto_t_COAP_PROTO_DTLS)
     }
 
+    // TODO: Possibly wrap in feature?
+    /// Joins a multicastgroup
+    /// # Safety
+    /// The libcoap function does not modify/delete any variables here.
+    /// NULL for the interface is handled internally in the libcoap function, therefore safe.
+    pub fn join_mcast_group_intf(
+        &mut self,
+        groupname: &str,
+        ifname: Option<&str>,
+    ) -> Result<(), MulticastGroupJoinError> {
+        let inner_ref = self.inner.borrow();
+        let mcast_groupname = CString::new(groupname).expect("CString::new failed");
+        let mcast_ifname = ifname.and_then(|if_in| Some(CString::new(if_in).expect("CString::new failed")));
+        let mcast_ifname_ptr = match mcast_ifname {
+            Some(if_in) => if_in.as_ptr(),
+            None => std::ptr::null(),
+        };
+        unsafe {
```

Comment:
Safety comments should only be part of the external method documentation if the method itself is unsafe (following the pattern [used in the standard library](https://std-dev-guide.rust-lang.org/policy/safety-comments.html)).
Instead, the safety comment should be added as a regular comment directly above the unsafe code block.
 
```suggestion
    pub fn join_mcast_group_intf(
        &mut self,
        groupname: &str,
        ifname: Option<&str>,
    ) -> Result<(), MulticastGroupJoinError> {
        let inner_ref = self.inner.borrow();
        let mcast_groupname = CString::new(groupname).expect("CString::new failed");
        let mcast_ifname = ifname.and_then(|if_in| Some(CString::new(if_in).expect("CString::new failed")));
        let mcast_ifname_ptr = match mcast_ifname {
            Some(if_in) => if_in.as_ptr(),
            None => std::ptr::null(),
        };
        
        // SAFETY: `inner_ref.raw_context` is always valid by construction of this type, group and interface name are pointers to valid C strings. 
        unsafe {
```